scripts/typemock/src/arg.ts

```diff
@@ -0,0 +1,298 @@
+/**
+ * Represents an argument condition used during verification.
+ */
+export class Arg {
+    private _validate: (value: any) => boolean;
+    private _message: string;
+    private _rest: boolean;
+
+    private constructor(condition: (value: any) => boolean, message: string, rest = false) {

```

Comment:
Why not just use parameter properties for `_validate`, `_message`, and `_rest` here?

Reply:
I'm personally not a fan of parameter properties and generally avoid using them.